src/System.Runtime.Extensions/tests/System/IO/PathTests.cs

```diff
@@ -131,24 +136,30 @@ public static void GetDirectoryName_Windows(string path, string expected)
         public static void GetDirectoryName_Unix(string path, string expected)
         {
             Assert.Equal(expected, Path.GetDirectoryName(path));
+            Assert.Equal(expected ?? string.Empty, new string(Path.GetDirectoryName(path.AsReadOnlySpan())));
         }
 
         [Fact]
         public static void GetDirectoryName_CurrentDirectory()
         {
             string curDir = Directory.GetCurrentDirectory();
             Assert.Equal(curDir, Path.GetDirectoryName(Path.Combine(curDir, "baz")));
+            Assert.Equal(curDir, new string(Path.GetDirectoryName(Path.Combine(curDir, "baz").AsReadOnlySpan())));
+
             Assert.Equal(null, Path.GetDirectoryName(Path.GetPathRoot(curDir)));
+            Assert.Equal(string.Empty, new string(Path.GetDirectoryName(Path.GetPathRoot(curDir).AsReadOnlySpan())));
         }
 
         [PlatformSpecific(TestPlatforms.AnyUnix)]  // Checks Unix-specific special characters in directory path
-        [Fact]
-        public static void GetDirectoryName_ControlCharacters_Unix()
+        [Theory]
+        [InlineData('\t', 1, "file")]
+        [InlineData('\b', 2, "fi le")]
+        [InlineData('\v', 3, "fi\nle")]
+        [InlineData('\n', 4, "fi\rle")]
+        public static void GetDirectoryName_ControlCharacters_Unix(char ch, int count, string file)
         {
-            Assert.Equal(new string('\t', 1), Path.GetDirectoryName(Path.Combine(new string('\t', 1), "file")));
-            Assert.Equal(new string('\b', 2), Path.GetDirectoryName(Path.Combine(new string('\b', 2), "fi le")));
-            Assert.Equal(new string('\v', 3), Path.GetDirectoryName(Path.Combine(new string('\v', 3), "fi\nle")));
-            Assert.Equal(new string('\n', 4), Path.GetDirectoryName(Path.Combine(new string('\n', 4), "fi\rle")));
+            Assert.Equal(new string(ch, count), Path.GetDirectoryName(Path.Combine(new string(ch, count), file)));
+            Assert.Equal(new string(ch, count), new string(Path.GetDirectoryName(Path.Combine(new string(ch, count), file).AsReadOnlySpan())));
```

Comment:
is the second `new string` required -- I would expect the compiler to implicitly cast the first parameter to ROS<char> so Assert.Equal could work on two of those.

Reply:
same elsewhere

Reply:
@danmosemsft  as an error when I try to do this 
```
The type 'ReadOnlySpan<char>' may not be used as a type argument
```

Reply:
@ahsonkhan is that expected, that this method Assert<T>(T one, T two) cannot accept ROS<char> as a type?

Reply:
Yes, that is expected. As far as I know, span (or any by-ref struct) cannot be passed as an argument to a generic method which results in generic instantiation.

See example test case here:
https://github.com/dotnet/coreclr/pull/15746/files#diff-15a8cde4e37759a5dde2eeadf7b81994R75

cc @VSadov

Reply:
Here is more details on Span safety:
https://github.com/dotnet/csharplang/blob/master/proposals/csharp-7.2/span-safety.md#language-constraints

> A `ref struct` type may not be used as a type argument, or as an element type in a tuple type.

Reply:
OK, thanks @ahsonkhan